.github/workflows/api-test.yml

```diff
@@ -44,9 +52,9 @@ jobs:
               - '!(docs/**)'
   build:
     name: API-Test-Build
-    needs: paths-filter
+    needs: [ paths-filter, get-runner ]
     if: ${{ (needs.paths-filter.outputs.not-ignore == 'true') || (github.event_name == 'push') }}
-    runs-on: ubuntu-latest
+    runs-on: ${{needs.get-runner.outputs.RUNNER}}
```

Comment:
There is no guarantee that the runner that runs previous steps can be picked up again in this step.